tasks/main.yml

```diff
@@ -15,12 +13,29 @@
       paths:
         - "{{ role_path }}/vars"
 
+- name: Find all files that could contain Debian package repos
+  block:
+    - name: Find /etc/apt/sources.list
+      ansible.builtin.find:
+        follow: yes
+        paths:
+          - /etc/apt
+        patterns:
+          - sources.list
+      register: sources_list
+    - name: Find all files in /etc/apt/sources.list.d
```

Comment:
This is a substantial change in functionality to slip in through a Lineage PR. I am also not a fan because a system may be configured with non-HTTPS supporting third-party apt repositories and this would break such configurations.

Reply:
I didn't change it just for the sake of changing it.  The old code was broken because Debian 12 now does away with `/etc/apt/sources.list` and instead uses `/etc/apt/sources.list.d/debian.sources`.

It is fairly common for unenlightened folks to append new third-party `apt` repos onto `/etc/apt/sources.list` instead of creating a file in `/etc/apt/sources.list.d`, so the "this might break third-party `apt` repos" concern should be a concern with the original code as well.  Or am I wrong about that?

Reply:
OK, I will rework the changes to make them less drastic.

Reply:
If you approve these changes then I will drop commits 3657ebf and 64a11f4 and force push, just to clean up the history.

Reply:
Please see commit a45c4a7.

Reply:
Thoughts on [this approach](https://github.com/cisagov/ansible-role-apt-over-https/compare/lineage/skeleton...testing/test_alternate_bookworm_implementation) instead? Although not properly documented the source file list is a default intentionally. Moving it to a variable would I believe break the ability to specify different files when using this as a role.

Reply:
Please see commits f0e2f6a, 3d049f0, 0c90ca3, c3f3e7e, and 76017b6.  If you approve of those changes then I will go back and clean up the history of this PR.

Reply:
That group of commits looks good to me! Thanks for the cleanup and patience.